src/JSystem/J2D/J2DGrafContext.cpp

```diff
@@ -280,604 +96,176 @@ void J2DGrafContext::setup2D()
 
 /*
  * --INFO--
  * Address:	80036658
  * Size:	000034
  */
-void J2DGrafContext::place(const JGeometry::TBox2<float>&)
+void J2DGrafContext::place(const JGeometry::TBox2f& bounds)
 {
-	/*
-	lwz      r7, 0(r4)
-	lwz      r6, 4(r4)
-	stw      r7, 4(r3)
-	lwz      r5, 8(r4)
-	stw      r6, 8(r3)
-	lwz      r0, 0xc(r4)
-	stw      r5, 0xc(r3)
-	stw      r0, 0x10(r3)
-	stw      r7, 0x14(r3)
-	stw      r6, 0x18(r3)
-	stw      r5, 0x1c(r3)
-	stw      r0, 0x20(r3)
-	blr
-	*/
+	m_bounds        = bounds;
+	m_scissorBounds = bounds;
 }
 
 /*
  * --INFO--
  * Address:	8003668C
  * Size:	00011C
  */
-void J2DGrafContext::setColor(JUtility::TColor, JUtility::TColor, JUtility::TColor, JUtility::TColor)
+void J2DGrafContext::setColor(JUtility::TColor colorTL, JUtility::TColor colorTR, JUtility::TColor colorBR, JUtility::TColor colorBL)
 {
-	/*
-	.loc_0x0:
-	  lbz       r0, 0x0(r4)
-	  li        r9, 0x1
-	  lbz       r11, 0x1(r4)
-	  li        r8, 0x4
-	  stb       r0, 0x24(r3)
-	  li        r0, 0x5
-	  lbz       r10, 0x2(r4)
-	  stb       r11, 0x25(r3)
-	  lbz       r4, 0x3(r4)
-	  stb       r10, 0x26(r3)
-	  lbz       r10, 0x0(r5)
-	  stb       r4, 0x27(r3)
-	  lbz       r4, 0x1(r5)
-	  stb       r10, 0x28(r3)
-	  lbz       r10, 0x2(r5)
-	  stb       r4, 0x29(r3)
-	  lbz       r4, 0x3(r5)
-	  stb       r10, 0x2A(r3)
-	  lbz       r5, 0x0(r6)
-	  stb       r4, 0x2B(r3)
-	  lbz       r4, 0x1(r6)
-	  stb       r5, 0x2C(r3)
-	  lbz       r5, 0x2(r6)
-	  stb       r4, 0x2D(r3)
-	  lbz       r4, 0x3(r6)
-	  stb       r5, 0x2E(r3)
-	  lbz       r5, 0x0(r7)
-	  stb       r4, 0x2F(r3)
-	  lbz       r4, 0x1(r7)
-	  stb       r5, 0x30(r3)
-	  lbz       r5, 0x2(r7)
-	  stb       r4, 0x31(r3)
-	  lbz       r4, 0x3(r7)
-	  stb       r5, 0x32(r3)
-	  stb       r4, 0x33(r3)
-	  stb       r9, 0xB0(r3)
-	  stb       r8, 0xB1(r3)
-	  stb       r0, 0xB2(r3)
-	  stb       r9, 0xB3(r3)
-	  stb       r8, 0xB4(r3)
-	  stb       r0, 0xB5(r3)
-	  stb       r9, 0xB6(r3)
-	  stb       r8, 0xB7(r3)
-	  stb       r0, 0xB8(r3)
-	  lwz       r0, 0x24(r3)
-	  rlwinm    r0,r0,0,24,31
-	  cmplwi    r0, 0xFF
-	  bnelr-
-	  li        r4, 0
-	  stb       r4, 0xB0(r3)
-	  stb       r9, 0xB1(r3)
-	  stb       r4, 0xB2(r3)
-	  lwz       r0, 0x2C(r3)
-	  rlwinm    r0,r0,0,24,31
-	  cmplwi    r0, 0xFF
-	  bnelr-
-	  stb       r4, 0xB3(r3)
-	  stb       r9, 0xB4(r3)
-	  stb       r4, 0xB5(r3)
-	  lwz       r0, 0x28(r3)
-	  rlwinm    r0,r0,0,24,31
-	  cmplwi    r0, 0xFF
-	  bnelr-
-	  lwz       r0, 0x30(r3)
-	  rlwinm    r0,r0,0,24,31
-	  cmplwi    r0, 0xFF
-	  bnelr-
-	  stb       r4, 0xB6(r3)
-	  stb       r9, 0xB7(r3)
-	  stb       r4, 0xB8(r3)
-	  blr
-	*/
+	m_colorTL = colorTL;
+	m_colorTR = colorTR;
+	m_colorBR = colorBR;
+	m_colorBL = colorBL;
+
+	_B0        = J2DBlendInfo(1, 4, 5);
+	m_linePart = J2DBlendInfo(1, 4, 5);
+	m_boxPart  = J2DBlendInfo(1, 4, 5);
+
+	if ((u8)u32(m_colorTL) != 0xFF) {
+		return;
+	}
+
```

Comment:
thoughts on a u8 operator?

Reply:
I haven't seen this crop up yet outside of this one singular instance, so I'm not sure how useful/warranted it'd be. I think if it comes up again it's probably worth working in a u8 operator though? Worth keeping an eye on.